test/rewrite_test.rb

```diff
@@ -12,15 +12,28 @@ def mod_src_check_file f
 
   def mod_src_check src, filename = nil
     begin
-      RubyVM::AbstractSyntaxTree.parse(src)
+      # ideally we'd use Ruby's actual parser to check that the code we've
+      # generated is parseable, but that's not possible unless
+      # RubyVM::AbstractSyntaxTree.parse is defined, which is only on MRI
```

Comment:
FYI it is possible using e.g. https://github.com/ruby/irb/pull/134/files
But since the parser gem is already a dependency, it seems fine to use it.

Reply:
I didn't measure, but it seems faster.
Anyway, introducing `valid_syntax?(src)` intermediate method will be good idea.